fix: minor fix to display the correct icon on dark mode

lucascnunes opened the pull request:
I've tried to fix it without recurring into a workaround but when vitepress is built the bug remained, this way, at least, when the vitepress is built the correct icon remained.

_Displaying the bug:_
When changing the theme into dark mode and refreshing the page it displays the wrong icon.

https://github.com/radix-vue/shadcn-vue/assets/2693803/d83fe8c4-316d-4feb-852d-f15a9d3c418f

_The fix:_
With the workaround it "flashes" the wrong icon but the correct one remains.

https://github.com/radix-vue/shadcn-vue/assets/2693803/0dd0507a-ada4-4197-9b9f-e80ba291fbcf


If anyone find the reason for the glitch or have a better way to fix it let me know.

I wish my first help would be a better one but this icon glitch was bugging me 😆

lucascnunes committed:
fix: minor 'workaround' to display the correct icon after refreshing the page when in dark mode

sadeghbarati commented:
@lucascnunes Thanks!

Yes that Glitch is so annoying, Idk how Vitepress doc handle it right 🤔 

I think `ClientOnly` wrapper would be better approche

```html
<ClientOnly>
  <Button
    class="flex items-center justify-center"
    aria-label="Toggle dark mode"
    :variant="'ghost'"
    :size="'icon'" @click="toggleDark()"
  >
    <component
      :is="isDark ? RadixIconsSun : RadixIconsMoon"
      class="w-[20px] h-5 text-foreground"
    />
  </Button>
</ClientOnly>
```

lucascnunes commented:
I'm going to test it, and push a fix if it works.
Thanks @sadeghbarati 

All done, tested and fixed the fix 😅 and pushed the correction.
Thanks for the input and help @sadeghbarati

lucascnunes committed:
fix: fixing glitch on dark mode icon

fixing the glitch when reloading the page on dark mode and the icon changing back to light mode.
